### PR TITLE
Closes #436 (and #122?): Add options to disable smooth cables and physics

### LINKED ISSUE #436: Make the curvature of cables and connections adjustable
### NetBox version

3.6.6

### Topology Views version

3.1.0

### Feature type

Change to existing functionality

### Proposed functionality

Please make this feature adjustable, or turn it off completely. 

### Use case

When building projects with more than a few nodes it quickly becomes apparent, that the cables and connections curve towards the middle. This is a nice tool to find the middle of the workspace, but the farther you get from the middle the bigger the curve gets. And with less than 100 nodes it already looks weird on the more distant clusters. 
It's a small issue but making it adjustable makes the whole topology much more presentable. 

### External dependencies

_No response_

### PR DESCRIPTION
**Pull request by dremerb**
Although both issues has status "under review", this maybe makes the decision to accept the issue easier.

PR adds settings to make cable renderings straight lines, as proposed in #436.
Line style is controlled by an option in the "Individual Options".
I am currently working with ~200 hosts and 500+ cables, this change makes the graph a lot clearer in my opinion.

Two side effects:
- With physics enabled, there is a more movement in the graph - but I want to propose another PR that allows turning that off anyway, maybe that synergy works out fine
- Since cables are straight lines now, they are "bundled" (#122). Or rather plotted on top of each other. The cable's shadows overlay each other as well, which may be considered "not pretty", but accidentally works well as an indicator of multiple connections between two ends.

Fixing the shadows could be done extracting `edges.append(create_edge(...))` calls into a dedicated function and ignoring duplicate connections.

### CODE REVIEW

**Review by mattieserver — CHANGES_REQUESTED** [review]
Can you maybe change the  name of the vars you added?
Maybe make it 'straight_cables' or something like that.

**Review by mattieserver — APPROVED** [review]
Looks good, will merge after testing

**Review by dreng — CHANGES_REQUESTED** [review]
You have overlooked the saved filters.

api/views.py, line 112 and below
views.py, line 713 and below.

**Review by dreng — CHANGES_REQUESTED** [review]
You have not taken get_query_settings() into account in utils.py.

File: netbox_topology_views/api/views.py
Comment (dreng) [inline netbox_topology_views/api/views.py]: That is not the way we did it so far. Please be consistent. Otherwise the code is harder to read. In addition, this solution may make it impossible to override the individual option.

It is not wrong to optimize code in general, but this should be a seperate task and should be done as a whole.

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py]: That is not the way we did it so far. Please be consistent. Otherwise the code is harder to read. In addition, this solution may make it impossible to override the individual option.

It is not wrong to optimize code in general, but this should be a seperate task and should be done as a whole.

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py]: That is not the way we did it so far. Please be consistent. Otherwise the code is harder to read. In addition, this solution may make it impossible to override the individual option.

It is not wrong to optimize code in general, but this should be a seperate task and should be done as a whole.

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py]: That is not the way we did it so far. Please be consistent. Otherwise the code is harder to read. In addition, this solution may make it impossible to override the individual option.

It is not wrong to optimize code in general, but this should be a seperate task and should be done as a whole.

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py]: That is not the way we did it so far. Please be consistent. Otherwise the code is harder to read. In addition, this solution may make it impossible to override the individual option.

It is not wrong to optimize code in general, but this should be a seperate task and should be done as a whole.

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py]: That is not the way we did it so far. Please be consistent. Otherwise the code is harder to read. In addition, this solution may make it impossible to override the individual option.

It is not wrong to optimize code in general, but this should be a seperate task and should be done as a whole.

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py]: That is not the way we did it so far. Please be consistent. Otherwise the code is harder to read. In addition, this solution may make it impossible to override the individual option.

It is not wrong to optimize code in general, but this should be a seperate task and should be done as a whole.

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py:1152]: That is not the way we did it so far. Please be consistent. Otherwise the code is harder to read. In addition, this solution may make it impossible to override the individual option.

It is not wrong to optimize code in general, but this should be a seperate task and should be done as a whole.

**Review by dreng — CHANGES_REQUESTED** [review]

File: netbox_topology_views/forms.py
Comment (dreng) [inline netbox_topology_views/forms.py:511]: Why did you omit an entry in class DeviceFilterForm?

**Review by dreng — CHANGES_REQUESTED** [review]

File: netbox_topology_views/forms.py
Comment (bremerle) [reply netbox_topology_views/forms.py:511]: Sorry, don't understand this, could you elaborate where I removed the entry?

Comment (bremerle) [reply netbox_topology_views/forms.py:511]: Nevermind, figured it out. Added.

File: netbox_topology_views/views.py
Comment (bremerle) [reply netbox_topology_views/views.py:1152]: For this I need a pointer what the current way is. Is this fixed by changing line 797?

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py:241]: Shouldn't this be an optional parameter? I'd use the Type Hint Syntax like in the other parameters below.

File: netbox_topology_views/views.py
Comment (dreng) [inline netbox_topology_views/views.py:306]: If this is optional (see comment above), you need to check for 'None' like in the other ones above.

**Review by dreng — CHANGES_REQUESTED** [review]

File: netbox_topology_views/views.py
Comment (bremerle) [reply netbox_topology_views/views.py:241]: My reasoning behind making it mandatory was, that it will have a default value of False from `IndividualOptions`/`get_query_settings` anyway. To me, `circuit/cable/...` are optional, because one of those 5 will be set (it is an if-elif without an else in the code), while `straight_cables` will have a value in any case (unless you didnt run the migrations).  
I did not see a use case for mixing straight and smooth cables, therefore making it mandatory may reduce errors in the future. E.g. when a new connection type is introduced, `straight_cables` does not get passed and stays `False`, while all other cables get the value passed and are straight.

It feels like a style choice to me, but I'd be happy to make it optional as well, your call :)  
But instead of `None`, i propose to initialize as `False`, since that is the global default value (keeping the current style with smooth cables as default).

**Review by dreng — APPROVED** [review]

**Comment by dreng** [comment]
We need to finish compatibility with NetBox 4.0 first. This PR may need to be rebased after that.

**Comment by mattieserver** [comment]
I will try to test/merge this this weekend.

**Pull request merged**
